frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx

```diff
@@ -321,3 +305,25 @@ const maybeAssoc: typeof assoc = (collection, key, value) => {
 
   return assoc(collection, key, value);
 };
+
+// The approximate number of maximum characters for values could be around 7
+// because the values will be compact format when the average length of
+// compact format is less than the average length of full format by 3 characters.
+// https://github.com/metabase/metabase/blob/96e9febfb31d231e3ef08ae907fb42773e065ca5/frontend/src/metabase/static-viz/components/XYChart/Values/Values.tsx#L175
+// e.g. 110,000 vs 100K = 4 !< 7 - 3 it will use full format in this case.
+// So, we could estimate that 7 is the maximum number of characters for values.
+// Then 3.5 is the half the length of the approximate maximum value.
+const APPROXIMATE_MAX_VALUE_CHAR_LENGTH = 7;
+const MAX_SERIES_LENGTH = 15;
```

Comment:
This number is eye-balled.
#### chart with max data length = 15
![Screen Shot 2022-08-30 at 5 28 25 PM](https://user-images.githubusercontent.com/1937582/187415506-1d00edcc-87db-4827-9c46-ef4f4c842f39.png)


#### chart with max data length = 16
![Screen Shot 2022-08-30 at 5 28 37 PM](https://user-images.githubusercontent.com/1937582/187415549-b2e6f0ef-f4f0-472b-b99f-5e0af0d49e06.png)